app/lib/lets-encrypt.server.ts

```diff
@@ -29,8 +87,93 @@ class LetsEncrypt {
       directoryUrl: this.#directoryUrl,
       accountKey: this.#accountKey,
     });
+
+    await this.#registerReloadAccount();
+
+    return this;
+  };
+
+  /**
+   * This fn extracts and requests challenge information from / based on
+   * the authorizations array that we got in the order.
+   */
+  #extractChallenges = async () => {
+    this.#authorizations = await this.#client!.getAuthorizations(this.#order!);
+
+    /**
+     * Each authorization (a soon to be common name or alt name
+     * we requested in createOrder => identifiers) contains one or
+     * more challenges. At least one challenge per authorization needs
+     * to be passed for the authorization to be granted
+     *
+     * Later, we can use the granted authorizations in our certificate
+     */
+    const promises = this.#authorizations.map(async (authorization) => {
+      const selectedChallenge = authorization.challenges.find(
+        ({ type }) => type === 'dns-01'
+      ) as AcmeDnsChallenge;
+
+      // Get key for challenge
+      const keyAuthorization = await this.#client!.getChallengeKeyAuthorization(selectedChallenge!);
+
+      const challengeBundle: ChallengeBundle = {
+        domain: `_acme-challenge.${authorization.identifier.value}`,
+        value: keyAuthorization,
+      };
+
+      return challengeBundle;
+    });
+
+    this.#challengeBundles = await Promise.all(promises);
+
+    console.log(JSON.stringify(this.#challengeBundles));
```

Comment:
I removed the console.

What type of test you think we should do? I think it would make most sense to do an integration test against a dockerized let's encrypt when we have those parts ironed out. If we wanted to do a unit test, we'd have to mock out / stub a large amount of api calls. Still possible, but less than optimal imo.

Reply:
I agree, we should do integration tests with the Pebble test container.  We'll do it in follow-ups vs. here.

Reply:
Perfect, thank you!

Reply:
Added Pebble for this in https://github.com/Seneca-CDOT/starchart/pull/103